src/sql.ts

```diff
@@ -168,6 +180,20 @@ function buildSpreadUpdateFragment(
   }
 }
 
+/**
+ * Extract colum names from a records.
+ * Make sure all records has the same number of columns.
+ */
+function extractColumsName(records: ValueRecord[]) {
+  const referenceRecord = records[0]
+  if (records.some(record => Object.keys(record).length !== Object.keys(referenceRecord).length)) {
+    throw new Error("Rows must be of the same length")
+  }
```

Comment:
This error message will yield when the rows are of different size. eg.
```
sql`INSERT INTO users ${spreadInsert({
    name: "Hugo",
    age: 20,
    surname: "Jonson"
  }, {
    name: "Jon",
    age: 25
  })} RETURNING *`
``` 

I think it's a okay error message. Alternatively we can skip this check and letting row 148 check if we have a value or not. We would then get the following error instead: `TypeError: Missing value for column "surname"` which can be more readable but that would result that the following snippet don't throw any error:
```
sql`INSERT INTO users ${spreadInsert({
    name: "Hugo",
    age: 20
  }, {
    name: "Jon",
    age: 25,
    surname: "Jonson"
  })} RETURNING *`
``` 

I do not think it is likely that someone is trying to insert multiple rows in a database with different shape.

So I think it's better to keep this check.